starknet-foundry/crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -180,18 +276,36 @@ fn execute_syscall(
 
     let calldata = buffer.next_arr().unwrap();
 
+    let calldata_blockifier: Vec<blockifier_Felt252> = calldata
+        .into_iter()
+        .map(|v| convert_to_blockifier_felt(&v))
+        .collect();
```

Comment:
Not sure, but I think `&v` wouldn't be needed if we used `iter()` instead of `into_iter()`.

Reply:
Good catch 👍

Reply:
[Review 2](https://github.com/foundry-rs/starknet-foundry/pull/181/commits/177b7e3da6761ffb59c65ba1bb3b0f81211af4e2)